flask_lesson_advanced/blueprint/product_main.py

```diff
@@ -0,0 +1,54 @@
+from flask import Blueprint, render_template, request, session
+import os
+from werkzeug.utils import secure_filename
+from infractructure import DB
+from models import Product_class
+
+product = Blueprint('product', __name__, template_folder='template_product',
+                    static_folder='static_product')
+
+
+@product.route('/product', methods=['GET'])
+def products_fun():
+    arg_price = request.args.get("price")
+    products_list = [{"name": rest.name, "id": rest.id, "price": rest.price,
+                      "img_name": rest.img_name,
+                      "description": rest.description}
+                     for rest in DB['products']]
+    product_filter = [i for i in products_list if str(i["price"]) == str(arg_price)]
+    sess = session.keys()
+    if len(product_filter) > 0:
+        return render_template('all_products.html', value=product_filter, sess=sess)
+    return render_template('all_products.html', value=products_list, sess=sess)
+
+
+@product.route('/product/<id>', methods=['GET'])
+def product_id_fun(id):
+    products_list = [{"name": rest.name, "id": rest.id, "price": rest.price,
+                      "img_name": rest.img_name,
+                      "description": rest.description}
+                     for rest in DB['products']]
+    for product_dict in products_list:
```

Comment:
what if you cannot find any product?
return 404 pag at `for...else`

Reply:
done

Reply:
we had chat about it and I have to make a page with information message